### ISSUE TITLE
Doesn't work with Webpack 2

### ISSUE THREAD

**Issue by schickling** [smooch/smooch-js#357]
More and more people are using Webpack 2 and most starter kit are based on it nowadays. Unfortunately your package doesn't work with it.

Could you please either make it compatible or provide a prebuilt file?

**Comment by lemieux** [comment]
You can try using the prebuilt package we have on the CDN at https://cdn.smooch.io/smooch.min.js

**Comment by schickling** [comment]
@lemieux how would I use it via npm?

**Comment by lemieux** [comment]
I guess for now you can't, but you could vendor in that file and alias it with webpack. I'll add the prebuilt package npm suggestion to our backlog.

In the meantime, what is not working with Typescript 2?

**Comment by schickling** [comment]
Oh apologies. I actually meant Webpack 2. (Seems like I cannot control my excitement about Typescript 2 😄

**Comment by lemieux** [comment]
So, same question : what's not working with webpack 2? :)

**Comment by schickling** [comment]
![screen shot 2016-08-09 at 23 36 41](https://cloud.githubusercontent.com/assets/1567498/17534606/3716c7ec-5e8a-11e6-8a15-cdec4ba0ea86.png)

```
 [1751] ./~/websocket-extensions/lib/pipeline/index.js 1.48 kB {1} [built]
 [1752] ./~/websocket-extensions/lib/websocket_extensions.js 4.86 kB {1} [built]
 [1756] multi vendor 400 bytes {1} [built]
     + 2 hidden modules

ERROR in ./~/faye/node/faye-node.js
Module not found: Error: Can't resolve 'fs' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/faye/node'
 @ ./~/faye/node/faye-node.js 2729:13-26
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/faye/node/faye-node.js
Module not found: Error: Can't resolve 'net' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/faye/node'
 @ ./~/faye/node/faye-node.js 2732:13-27
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/faye/node/faye-node.js
Module not found: Error: Can't resolve 'tls' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/faye/node'
 @ ./~/faye/node/faye-node.js 2734:13-27
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/tunnel-agent/index.js
Module not found: Error: Can't resolve 'net' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/tunnel-agent'
 @ ./~/tunnel-agent/index.js 3:10-24
 @ ./~/faye/node/faye-node.js
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/tunnel-agent/index.js
Module not found: Error: Can't resolve 'tls' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/tunnel-agent'
 @ ./~/tunnel-agent/index.js 4:10-24
 @ ./~/faye/node/faye-node.js
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/tough-cookie/lib/cookie.js
Module not found: Error: Can't resolve 'net' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/tough-cookie/lib'
 @ ./~/tough-cookie/lib/cookie.js 32:10-24
 @ ./~/faye/node/faye-node.js
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/faye-websocket/lib/faye/websocket/client.js
Module not found: Error: Can't resolve 'net' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/faye-websocket/lib/faye/websocket'
 @ ./~/faye-websocket/lib/faye/websocket/client.js 2:13-27
 @ ./~/faye-websocket/lib/faye/websocket.js
 @ ./~/faye/node/faye-node.js
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor

ERROR in ./~/faye-websocket/lib/faye/websocket/client.js
Module not found: Error: Can't resolve 'tls' in '/Users/johanens/Desktop/work/graphcool/dashboard/node_modules/faye-websocket/lib/faye/websocket'
 @ ./~/faye-websocket/lib/faye/websocket/client.js 3:13-27
 @ ./~/faye-websocket/lib/faye/websocket.js
 @ ./~/faye/node/faye-node.js
 @ ./~/smooch/lib/utils/faye.js
 @ ./~/smooch/lib/services/conversation-service.js
 @ ./~/smooch/lib/smooch.js
 @ ./~/smooch/lib/main.js
 @ multi vendor
Child html-webpack-plugin for "index.html":
    chunk    {0} index.html 525 kB [entry] [rendered]
        [0] ./~/lodash/lodash.js 523 kB {0} [built]
        [1] (webpack)/buildin/global.js 506 bytes {0} [built]
        [2] (webpack)/buildin/module.js 548 bytes {0} [built]
        [3] ./~/html-webpack-plugin/lib/loader.js!./src/index.html 665 bytes {0} [built]
webpack: bundle is now VALID.
```

**Comment by lemieux** [comment]
Seems like Webpack is trying to build for node. I'm very surprised this doesn't work since I actually switch to webpack 2 in the f/code-splitting branch (https://github.com/smooch/smooch-js/blob/f/code-splitting/make-webpack-config.js) and it worked pretty well.

At this point, I'm starting to think it's more a webpack configuration problem than a Smooch problem.

**Comment by schickling** [comment]
I couldn't find the problem but I'm not too familiar with the assumptions this package makes in terms of configuring Webpack. Here is our `webpack.config.js`: https://github.com/graphcool/dashboard/blob/master/webpack.config.js

**Comment by lemieux** [comment]
Try having `target: 'web'` in your exported config maybe? Other than that I'm not sure how to help you with that. I'll close the issue for now and will reopen if it ends up being a smooch.js specific problem.

**Comment by schickling** [comment]
Unfortunately `target: 'web'` didn't solve the problem. Could you please reopen the issue until this is resolved?

**Comment by alavers** [comment]
Closed because it doesn't seem like a smooch-js issue. Happy to reopen if we discover that it is a bug on our side.

What error are you now encountering with `target: 'web'`? Is it still telling you `Can't resolve 'fs'` ?

**Comment by schickling** [comment]
Yes, exactly the same problem.

**Comment by lemieux** [comment]
After investigating, it seems like webpack 2 doesn't use the `browser` field in subdependencies (hence why it's trying to use faye-node instead of faye-browser). For now, you could work around this problem by adding an alias to your webpack config like so `faye: 'faye/browser/faye-browser'. We might have a way to fix that right in the smooch package for the future.

Until we test and release that, the alias workaround should be fine.

**Comment by lemieux** [comment]
You only need to alias faye for the workaround.

**Comment by schickling** [comment]
Thanks @lemieux. That works. Please ping me when there is a fix for this :)

**Comment by lemieux** [comment]
This fix was release as part of 3.4.0.
